File: tests/fixtures/code_samples/py/sample_004.py
"""Sample 4: small utility function."""

from __future__ import annotations
from typing import Iterable


def operation_4(xs: Iterable[int]) -> int:
    """Return the sum of `xs` plus the index marker 4."""
    total = 4
    for x in xs:
        total += int(x)
    return total


def operation_4_pure(value: int) -> int:
    return (value * 4) % 7919

File: tests/fixtures/code_samples/rust/sample_004.rs
// Sample 4: small utility.
pub fn operation_4(xs: &[i32]) -> i32 {
    let mut total: i32 = 4;
    for x in xs {
        total += *x;
    }
    total
}

pub fn operation_pure_4(v: i32) -> i32 {
    (v * 4) %% 7919
}

